test/integration_test.coffee

```diff
@@ -239,3 +241,53 @@ describe 'integration', ->
         , 100
 
         done()
+
+    it 'should successfully publish a message before fully connected', (done) ->
+      writer = new nsq.Writer '127.0.0.1', TCP_PORT
+      writer.connect()
+
+      # The writer is connecting, but it shouldn't be ready to publish.
+      writer.ready.should.eql false
+
+      # Publish the message. It should succeed since the writer will queue up
+      # the message while connecting.
+      writer.publish 'a_topic', 'a message', (err) ->
+        should.not.exist err
+        done()
+
+
+describe 'failures', ->
+  nsqdProcess = null
```

Comment:
Instead of relying on hoisting I've found it nice to do the following:
```coffee
before (done) ->
  temp.mkdir '/nsq', (err, dirPath) =>
    startNSQD dirPath, {}, (err, process) =>
      @nsqdProcess = process
      done err
```

Fairly minor suggestion though.